lib/file-source/src/file_server.rs

```diff
@@ -505,3 +506,11 @@ impl Default for TimingStats {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct Line {
+    pub text: Bytes,
+    pub filename: String,
```

Comment:
Maybe a `Box<str>` since we'll have a lot of these running around?

Reply:
I'd be willing to entertain that, but I was aiming to minimize the changes to the file server component. For instance, that should really be a "batch" object, with a vector of lines for the one filename, but that's a job for another issue.

Reply:
Makes sense to me. With regard to the vector of lines, agree. I'm fairly nervous about the performance implications of this PR since the changes here are obligatory whether you have acks flipped on or not and moving lines out in a batch would address that concern.

Reply:
But, as you say, that's a job for another issue.

Reply:
This new `struct Line` is just putting a name on the `(Bytes, String)` tuple that the file server used to emit. As such, these changes should have no measurable performance impact on the default no-acknowledgements configuration.

Reply:
Oh, I take that back, the new struct also adds the file ID and offset. Still, those are both `Copy` types, so performance impact should be minimal.

Reply:
Right you are.

Reply:
Yep, agreed. My concern was allocating per-line but, like you say, that's unchanged in this PR.